config/config.go

```diff
@@ -311,7 +317,15 @@ func (tree *Tree) Root() *Root {
 	if tree.Parent != nil {
 		return tree.Parent.Root()
 	}
-	return NewRoot(tree)
+	return tree.root
```

Comment:
This change is needed because otherwise a new instance of root is given whenever you cann `tree.Root()`, which then loses the `hasTerragruntStacks` bool info.